src/common/file.c

```diff
@@ -590,6 +592,18 @@ util_file_open(const char *path, size_t *size, size_t minsize, int flags)
 			*size = (size_t)actual_size;
 			LOG(4, "actual file size %zu", *size);
 		}
+
+		int ret;
+		if (Fallc && (flags & O_RDWR)) {
+			if ((ret = os_posix_fallocate(fd, 0,
```

Comment:
Does this work with device dax?
Also, shouldn't we be checking if this is actually open with `O_CREAT | O_EXCL` flags?